### PR TITLE
small generating set

### PR DESCRIPTION
**Pull request by afkafkafk13**
- make mstd-functionality available for multivariate polynomial rings and quotients thereof
  (naming: small generating set, because we do not have minimality here!)
- make small_generating_set available for localizations at powers of elements
- make small_generating_set available on charts for IdealSheaves 
- use small_generating_set in example K3.jl

### CODE REVIEW

**Comment by afkafkafk13** [comment]
@HechtiDerLachs , @simonbrandhorst : This should do, what you wanted to achieve by calling radical in K3.jl (even for ideals known to already be radical). Please have a look.

**Comment by simonbrandhorst** [comment]
Since this terminology in principle makes sense also for ideals which are not multivariate, I cc. @fieker

**Comment by afkafkafk13** [comment]
> Since this terminology in principle makes sense also for ideals which are not multivariate, I cc. @fieker

From that perspective, I would suggest:
- small_generating_set also exists for graded rings as an alias for minimal_generating_set (to avoid confusion for users -- each of the two is available whenever it makes sense)  -- this will happen in this PR, if you agree
- dito for the local case, as soon as it is finished/merged -- on my plate, but not here, waiting for PR#2327
- I see univariate polynomial rings in a different context compared to multivariate ones (more precisely, they are euclidean rings), so it is up to @fieker to decide whether he also provides the two function names as aliases there -- independent of the current PR

**Comment by simonbrandhorst** [comment]
What I mean to say is that `small_generating_set` makes sense for almost any algebraic structure that has generators.
Groups, ideals, modules, algebras,  algebras, non-commutative algebras, vector spaces etc.
So I want to make sure we are reasonably consistent here.
The common definition of a `small_generating_set` would be: 1. It is a generating set. 2. It is "fast" to compute 3. It is "small", but in general we do not give any guarantees (except maybe that it should be at least as small as the input?) 
The second question: 
`minimal_generating_set` minimal with respect to inclusion or cardinality? I think for finite groups it is cardinality.

**Comment by afkafkafk13** [comment]
> What I mean to say is that `small_generating_set` makes sense for almost any algebraic structure that has generators. Groups, ideals, modules, algebras, algebras, non-commutative algebras, vector spaces etc. So I want to make sure we are reasonably consistent here. The common definition of a `small_generating_set` would be: 1. It is a generating set. 2. It is "fast" to compute 3. It is "small", but in general we do not give any guarantees (except maybe that it should be at least as small as the input?) 

Indeed, this is precisely the idea -- including your statement about at least as small as the input!

> The second question: `minimal_generating_set` minimal with respect to inclusion or cardinality? I think for finite groups it is cardinality.

Minimal is meant as in minimal free resolution in the local and graded case, where we have uniqueness of Betti numbers. 
So it has a standard mathematical definition here, which coincides with cardinality (1st Betti number of R/I).

**Comment by afkafkafk13** [comment]
Aside: tests are in doctests, not in the test-directory. So they might not show up correctly in the codecov (in particular for the quotient ring case).

File: experimental/Schemes/IdealSheaves.jl
Comment (simonbrandhorst) [inline experimental/Schemes/IdealSheaves.jl]: Please fix the indentation. Also the `Oscar.` should not be necessary.

File: src/Rings/MPolyQuo.jl
Comment (simonbrandhorst) [inline src/Rings/MPolyQuo.jl]:
```suggestion
  @req coefficient_ring(Q) isa Field "The coefficient ring must be a field"
```

File: src/Rings/MPolyQuo.jl
Comment (simonbrandhorst) [inline src/Rings/MPolyQuo.jl]:
```suggestion
small_generating_set(I::MPolyQuoIdeal{<:MPolyDecRingElem}) = minimal_generating_set(I)
```

**Review by simonbrandhorst — APPROVED** [review]

File: experimental/Schemes/IdealSheaves.jl
Comment (afkafkafk13) [reply experimental/Schemes/IdealSheaves.jl]: done.

File: src/Rings/mpoly-ideals.jl
Comment (simonbrandhorst) [inline src/Rings/mpoly-ideals.jl]:
```suggestion
  @req coefficient_ring(R) isa Field "The coefficient ring must be a field"
```

**Pull request merged**
